syncd/Ipc.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+#include <unistd.h>
+
+extern "C" {
+#include "sai.h"
+}
+
+#include "swss/logger.h"
+#include "swss/selectable.h"
+#include "meta/SaiInterface.h"
+
+namespace syncd
+{
+    class RpcStub;
+
+    enum {
+        IPC_TYPE_MIN = 0,
+        IPC_TYPE_REQ = IPC_TYPE_MIN, // request
+        IPC_TYPE_RESP,  // response
+        IPC_TYPE_NOTIF, // notification
+        IPC_TYPE_MAX,
+    };
+
+    enum {
+        IPC_SUBTYPE_MIN = 0,
+        IPC_SUBTYPE_MDIO = IPC_SUBTYPE_MIN, // mdio rpc
+        IPC_SUBTYPE_MAX,
+    };
+
+    struct IpcMsg {
+        uint16_t type;
+        uint16_t subtype;
+        uint32_t length;
+        char data[0];
+    };
+
+    static constexpr uint32_t IPC_MSG_MIN_LEN = sizeof(IpcMsg);
+    static constexpr uint32_t IPC_MSG_MAX_LEN = 1024;
+    static constexpr uint32_t IPC_TIMEOUT_SEC = 2;    // in seconds
+    static constexpr char IPC_SERVER[] = "/var/run/redis/syncd.sock";
+
+    class Ipc: public swss::Selectable
+    {
+    public:
+
+        Ipc(int pri = 0);
+
+        virtual ~Ipc() {
+            if (m_sock >= 0)
+            {
+                close(m_sock);
+            }
+        }
+
+        std::shared_ptr<RpcStub> rpcAt(int rpc) const;
+        void registerRpc(const std::shared_ptr<RpcStub> &stub);
+
+        int sendto(const IpcMsg *msg, const struct sockaddr *dest_addr, socklen_t addrlen);
+        int recvfrom(IpcMsg *msg, int size, struct sockaddr *src_addr, socklen_t *addrlen);
+        int send(const IpcMsg *msg);
+        int recv(IpcMsg *msg, int size);
```

Comment:
all pamameters inlcuding constructor should be 1 per line and have prefix specifier In or Out etc, take a look at ohter files .h